### ISSUE TITLE
Joule will not work on '2nd' PC

### ISSUE THREAD

**Issue by robclark56** [wbobeirne/joule-extension#183]
### Description of the Issue
* Setup Joule easily and successfully on PC1
* Joule on PC1 correctly connects to and displays LND node info
* With the magic of Google accounts, went to my 'other' PC and was pleasantly surprised to see Joule was magically already installed on the 2nd PC, and that the readonly.macaroon was installed. 
* But Joule was failing to contact my LND node.
* Everytime I go to edit the macaroons and paste my HEX admin.macaroon, it seems to succeed but the 2nd PC can not connect to my LND node

### Steps to Reproduce
On 2nd PC:
* Open Chrome -> Joule
* Settings
* See 
 Admin Macaroon = (encrypted)
 Readonly Macaroon = (Correct value) 
* Click [Edit] icon
* Click 'switch' to hex strings
* See 
 Admin Macaroon = paste hex string here
 Readonly Macaroon = (Correct value. Same as on PC1) 
* get my hex string from my lnd node with this command:
sudo xxd -ps -u -c 1000 /home/bitcoin/.lnd/data/chain/bitcoin/mainnet/admin.macaroon
* Paste the hex into admin.macaroon field (BTW .. same as used on PC1)
* Continue
* Enter Password
* Green message "Macaroons Updated' (or similar)
* Settings page show correct admin & readonly macaroons
* Back to main Joule page
* (!) Failed to fetch channels 
* try again no help

#### PC1 = Desktop 
* Windows: 10, 64-bit
* Chrome:  72.0.3626.121
* Processor: Intel i5, 

#### PC2 = Tablet 
* Windows 10, 64-bit
* Chrome:  72.0.3626.121
* Processor: Intel Atom x5-X8300

#### LND Node 
V0.5.2, running on Raspberry Pi

####  Network ####
PC1, PC2, and LND node all on same LAN (no routers between them)
LND has exposed REST interface on port 8082 (NON STANDARD  ... is that a clue???) 
```
# /home/bitcoin/.lnd/lnd.conf
[Application Options]
restlisten=10.1.1.141:8082
```
The only firewall is on the LND node, but it has these wide open settings for port 8082
```
admin ~  ฿  sudo ufw status | grep 8082
8082                       ALLOW       Anywhere                   # allow REST from WAN
8082                       ALLOW       10.1.1.0/24                # allow REST from LAN
8082 (v6)                  ALLOW       Anywhere (v6)              # allow REST from WAN
```
### Console Logs / Screenshots
![Capture2](https://user-images.githubusercontent.com/12369926/54068438-bf5be680-4287-11e9-9b6c-4bd3e72359d8.PNG)
![Capture](https://user-images.githubusercontent.com/12369926/54068439-bf5be680-4287-11e9-9828-fb9898542222.PNG)

**Comment by jamaljsr** [comment]
I have been testing this same setup on my two machines (Win10 & MacOS) and the syncing works without issue for me. I tried following your steps as well as a few other scenarios, such as resetting the connection on win10 and confirming it resets on the mac. Then connecting to my node from the mac and confirming the win10 machine is now connected. It even works when I used a different password after resetting the connection.

What I believe is happening in your case, it that PC2 is failing to connect due to an SSL certificate error. Remember that you have to load your node url in a separate tab, then click advanced and proceed. The certificate acceptance does not get synced by Chrome across machines. Try to open the url `https://10.1.1.141:8082/v1/getinfo` in a new tab on PC2 and confirm that you can see the message `expected 1 macaroon, got 0`.

Side note: we have PR #180 open which helps suggest fixes when there are connection problems, such as this certificate issue. Hopefully we can get this included in the next release.

**Comment by robclark56** [comment]
Problem Resolved. Thanks. Here is how it was fixed.

1) On PC2: Went to https://10.1.1.141:8082/v1/getinfo
2) Chrome warned that it was insecure. Advanced -> Proceed to 10.1.1.141 (unsafe) ... 
3) {"error":"expected 1 macaroon, got 0","code":2}
4) Now tried Joule on PC2 -> Success!

So ... seems that Joule was suffering the same https problem, but not reporting any diagnostic errors to the user. Only after I told Chrome to ignore https errors for 10.1.1.141, did Joule start working.

Why did I not see same issue on PC1? I have done other REST work from that PC to my LND node in the past and probably many months ago bypassed the https security check.

As an aside: Why was there a cert issue? Because it self-signed, and there was no trusted root:
**This CA Root certificate is not trusted because it is not in the Trusted Root Certification Authorities store.**
Issuer:   CN = RaspiBolt, O = lnd autogenerated cert
Valid from: ‎Tuesday, ‎24 ‎April ‎2018 12:53:56 PM  
Valid to: ‎Wednesday, ‎19 ‎June ‎2019 12:53:56 PM

So: Again, thank for giving me the clues to resolve. But this same issue will likely hit others in the same boat. Suggest catching such issues and giving user the tools/info to resolve.

**Comment by wbobeirne** [comment]
Thanks for clearing it up @jamaljsr, I've merged #180 in so this should no longer be an issue with the next release.

**Comment by wbobeirne** [comment]
@robclark56 to answer your question, self-signed certificates are typically rejected by browsers. LND can generate LetsEncrypt certs for domains, but unfortunately browsers don't like the certs signed for local usage. At the moment there's not a way around this using only a Chrome extension, it would require some kind of native proxy to fix it.
